src/libs/logging/src/writer.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2020 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::io::{Result, Write};
+
+use slog::{info, Logger};
+
+/// Writer to convert each line written to it to a log record.
+#[derive(Debug)]
+pub struct LogWriter(Logger);
+
+impl LogWriter {
+    /// Create a new isntance of ['LogWriter'].
+    pub fn new(logger: Logger) -> Self {
+        LogWriter(logger)
+    }
+}
+
+impl Write for LogWriter {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        buf.split(|b| *b == b'\n').for_each(|it| {
+            if !it.is_empty() {
+                info!(self.0, "{}", String::from_utf8_lossy(it))
+            }
+        });
+
+        Ok(buf.len())
+    }
+
+    fn flush(&mut self) -> Result<()> {
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{create_logger, FileRotator};
+    use std::fs;
+
+    #[test]
+    fn test_rotator_rotate() {
```

Comment:
AFAICT this doesn't really test anything about rotation.

Reply:
Changed to "test_log_writer()".